dll/cpl/desk/desktop.c

```diff
@@ -187,24 +267,55 @@ SetDesktopSettings(PDESKTOP_DATA pData)
         pData->bHideChanged[i] = FALSE;
     }
 
+    for (i = 0; i < _countof(IconChange); i++)
+    {
+        TCHAR iconPath[MAX_PATH];
+        DWORD dwType = (pData->Icon[i].szPath[0] == TEXT('%') ? REG_EXPAND_SZ : REG_SZ);
+
+        if (!pData->bIconChanged[i])
+            continue;
+
+        StringCchCopy(iconPath, _countof(iconPath), szUserClass);
+        StringCchCat(iconPath, _countof(iconPath), IconChange[i].CLSID);
+        StringCchCat(iconPath, _countof(iconPath), szDefaultIcon);
+
+        SHSetValue(HKEY_CURRENT_USER, iconPath, IconChange[i].Icon, dwType,
+                   pData->Icon[i].szPath, sizeof(pData->Icon[i].szPath));
+        if (IconChange[i].Title == IDS_TITLE_BIN_0)
```

Comment:
Just to confirm, that `.Title` contains here a numerical ID?

Reply:
Yes.